src/tscore/ThrottledMessage.cc

```diff
@@ -0,0 +1,96 @@
+/** @file
+
+  ThrottledMessage declaration.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "tscore/ThrottledMessage.h"
+
+ThrottledMessage::ThrottledMessage(std::chrono::seconds seconds_between)
+  : _last_log_time{TimePoint{}}, _seconds_between{seconds_between}
```

Comment:
What's the point of member initializing `_last_log_time`?

Reply:
Yeah, I was surprised I needed that, but without that explicit initialization the compiler complains with:

```
  CXX      ThrottledMessage.lo
ThrottledMessage.cc: In constructor 'ThrottledMessage::ThrottledMessage(std::chrono::seconds)':
ThrottledMessage.cc:27:37: error: use of deleted function 'constexpr std::atomic<_Tp>::atomic() [with _Tp = std::chrono::time_point<std::chrono::_V2::system_clock, std::chrono::duration<long int, std::ratio<1, 1000000000> > >]'
   : _seconds_between{seconds_between}
                                     ^
In file included from ../../include/tscore/ThrottledMessage.h:28,
                 from ThrottledMessage.cc:24:
/opt/rh/devtoolset-8/root/usr/include/c++/8/atomic:194:7: note: 'constexpr std::atomic<_Tp>::atomic() noexcept [with _Tp = std::chrono::time_point<std::chrono::_V2::system_clock, std::chrono::duration<long int, std::ratio<1, 1000000000> > >]' is implicitly deleted because its exception-specification does not match the implicit exception-specification ''
       atomic() noexcept = default;
       ^~~~~~
make: *** [ThrottledMessage.lo] Error 1